tools/gfx/d3d/d3d-util.cpp

```diff
@@ -583,4 +589,67 @@ bool D3DUtil::isUAVBinding(slang::BindingType bindingType)
+    IDebugCallback* callback)
+{
+    if (buildInputs.geometryDescs)
+    {
+        geomDescs.setCount(buildInputs.descCount);
+        for (Index i = 0; i < geomDescs.getCount(); i++)
+        {
+            auto& inputGeomDesc = buildInputs.geometryDescs[i];
+            geomDescs[i].Flags = translateGeometryFlags(inputGeomDesc.flags);
+            switch (inputGeomDesc.type)
+            {
+            case IAccelerationStructure::GeometryType::Triangles:
+                geomDescs[i].Type = D3D12_RAYTRACING_GEOMETRY_TYPE_TRIANGLES;
+                geomDescs[i].Triangles.IndexBuffer = inputGeomDesc.content.triangles.indexData;
+                geomDescs[i].Triangles.IndexCount = inputGeomDesc.content.triangles.indexCount;
+                geomDescs[i].Triangles.IndexFormat =
+                    D3DUtil::getMapFormat(inputGeomDesc.content.triangles.indexFormat);
+                geomDescs[i].Triangles.Transform3x4 = inputGeomDesc.content.triangles.transform3x4;
+                geomDescs[i].Triangles.VertexBuffer.StartAddress =
+                    inputGeomDesc.content.triangles.vertexData;
+                geomDescs[i].Triangles.VertexBuffer.StrideInBytes =
+                    inputGeomDesc.content.triangles.vertexStride;
+                geomDescs[i].Triangles.VertexCount = inputGeomDesc.content.triangles.vertexCount;
+                geomDescs[i].Triangles.VertexFormat =
+                    D3DUtil::getMapFormat(inputGeomDesc.content.triangles.vertexFormat);
+                break;
+            case IAccelerationStructure::GeometryType::ProcedurePrimitives:
+                geomDescs[i].Type = D3D12_RAYTRACING_GEOMETRY_TYPE_PROCEDURAL_PRIMITIVE_AABBS;
+                geomDescs[i].AABBs.AABBCount = inputGeomDesc.content.proceduralAABBs.count;
+                geomDescs[i].AABBs.AABBs.StartAddress = inputGeomDesc.content.proceduralAABBs.data;
+                geomDescs[i].AABBs.AABBs.StrideInBytes =
+                    inputGeomDesc.content.proceduralAABBs.stride;
+                break;
+            default:
+                callback->handleMessage(
+                    DebugMessageType::Error,
+                    DebugMessageSource::Layer,
+                    "invalid value of IAccelerationStructure::GeometryType.");
+                return SLANG_E_INVALID_ARG;
+            }
+        }
+    }
+    desc.DescsLayout = D3D12_ELEMENTS_LAYOUT_ARRAY;
+    desc.NumDescs = buildInputs.descCount;
+    switch (buildInputs.kind)
+    {
+    case IAccelerationStructure::Kind::TopLevel:
+        desc.Type = D3D12_RAYTRACING_ACCELERATION_STRUCTURE_TYPE_TOP_LEVEL;
+        desc.InstanceDescs = buildInputs.instanceDescs;
+        break;
+    case IAccelerationStructure::Kind::BottomLevel:
+        desc.Type = D3D12_RAYTRACING_ACCELERATION_STRUCTURE_TYPE_BOTTOM_LEVEL;
+        desc.pGeometryDescs = geomDescs.getBuffer();
+        break;
+    }
+    desc.Flags = (D3D12_RAYTRACING_ACCELERATION_STRUCTURE_BUILD_FLAGS)buildInputs.flags;
```

Comment:
If the external API has the requirement that it matches D3D12_RAYTRACING_ACCELERATION_STRUCTURE_BUILD_FLAGS we might want to put a comment to that effect in the interface and/or have a function that does the mapping.